fdbrpc/genericactors.actor.h

```diff
@@ -137,10 +137,15 @@ Future<REPLY_TYPE(Req)> retryGetReplyFromHostname(Req request, Hostname hostname
 	// Suitable for use with hostname, where RequestStream is NOT initialized yet.
 	// Not normally useful for endpoints initialized with NetworkAddress.
 	state double reconnetInterval = FLOW_KNOBS->HOSTNAME_RECONNECT_INIT_INTERVAL;
+	state RequestStream<Req>* to;
 	loop {
 		NetworkAddress address = wait(hostname.resolveWithRetry());
-		RequestStream<Req> to(Endpoint::wellKnown({ address }, token));
-		state ErrorOr<REPLY_TYPE(Req)> reply = wait(to.tryGetReply(request));
+		if (to == nullptr) {
+			to = new RequestStream<Req>(Endpoint::wellKnown({ address }, token));
```

Comment:
Will this memory be leaked if it's overwritten in a future loop? Why is this assigned using `new` and the else case isn't?

Reply:
Great catch! Fixed.